woocommerce/woocommerce
# Update payment gateway logic in payment task

### Changes proposed in this Pull Request:

Closes #32132.

- Adds logic to show only "additional" payment gateways when WCPay is supported and either WCPay or "other" payment gateway is set up.
- Applies the logic with the spreadsheet in paJDYF-3oZ-p2#comment-10304
- Adds 3 new gateways: Affirm, Amazon Pay, and Afterpay
- If a user has selected and set up either WCPay or a solution provided by “other payment providers”:
    - Only display in the online payment section the solutions included in the "Display as Additional Payment Option" category.
    - Hide the "See more" option at the bottom of the online payments section

### How to test the changes in this Pull Request:

**Test with eligible WCPay**:
1. In a fresh site, go through the setup wizard 
    - Select UK country but don't install WCpay
    - In business details, select "Yes, in person at physical stores"
3. Edit [this condition](https://github.com/woocommerce/woocommerce/blob/bc76636f3f1400731ef2e9eb36e5dcd313e51bed/plugins/woocommerce/src/Admin/Features/PaymentGatewaySuggestions/Init.php#L66) to true so that it returns the default gateways.
4. Go to "Set up payments" task
5. Click on "Other payment gateways"
6. Observe Stripe, Paypal, Mollie, Square (other category) gateways are shown above "Offline payment methods"
7. Install this WCPay plugin:  [woocommerce-payments-no-setup.zip](https://github.com/woocommerce/woocommerce/files/8479578/woocommerce-payments-no-setup.zip). This plugin emulates WCPay with `needsSetup = false`.
1. Refresh and observe the gateways are replaced with Klarna and Paypal (additional category).

**Test with WCPay ineligible**
1. Deactivate WCPay plugin
1. Change store country to Norway
4. Go to "Set up payments" task
5. Observe plugins listed: Stripe, PayPal, Klarna Checkout, Klarna Payments (other & additional category)
6. Install this PayPal plugin: [woocommerce-paypal-payments-no-setup.zip](https://github.com/woocommerce/woocommerce/files/8485948/woocommerce-paypal-payments-no-setup.zip)
7. Refresh and observe only Klarna Payments and Paypal is shown (additional category)

**Test ribbon change**
1. Deactivate Paypal plugin
1. Change store country to Mexico
4. Go to "Set up payments" task
5. Observe "Local Partner" pill next to Mercado Pago

<img width="730" alt="image" src="https://user-images.githubusercontent.com/3747241/163306034-f53ddaf5-539e-4b51-9d0f-7f1d09a0e5a0.png">

### Other information:




### Changelog entry

> Update payment gateway logic in payment task

### FOR PR REVIEWER ONLY:
